qiskit_nature/circuit/library/ansaetze/evolved_operator_ansatz.py

```diff
@@ -0,0 +1,135 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The evolved operator ansatz."""
+
+from typing import List, Union, Optional
+
+from qiskit.circuit import Parameter, ParameterVector, QuantumRegister
+from qiskit.circuit.library import BlueprintCircuit
+from qiskit.opflow import OperatorBase, EvolutionBase, PauliTrotterEvolution
+
+
+class EvolvedOperatorAnsatz(BlueprintCircuit):
+    """The evolved operator ansatz."""
+
+    def __init__(self,
+                 operators: Union[OperatorBase, List[OperatorBase]],
+                 reps: int = 1,
+                 evolution: Optional[EvolutionBase] = None,
+                 insert_barriers: bool = False,
+                 name: str = 'EvolvedOps') -> None:
+        """
+        Args:
+            operators: The operators to evolve.
+            reps: The number of times to repeat the evolved operators.
+            evolution: An opflow converter object to construct the evolution.
+                Defaults to Trotterization.
+            insert_barriers: Whether to insert barriers in between each evolution.
+            name: The name of the circuit.
+        """
+        if evolution is None:
+            evolution = PauliTrotterEvolution()
+
+        super().__init__(name=name)
+        self._operators = None
+        self._evolution = evolution
+        self._reps = reps
+        self._insert_barriers = insert_barriers
+
+        # use setter to set operators
+        self.operators = operators
+
+    def _check_configuration(self, raise_on_failure: bool = True) -> bool:
+        if self.operators is None:
+            if raise_on_failure:
+                raise ValueError('The operators are not set.')
+            return False
+
+        if self.reps < 0:
+            if raise_on_failure:
+                raise ValueError('The reps cannot be smaller than 0.')
+            return False
+
+    @property
+    def reps(self) -> int:
+        """The number of times the evolved operators are repeated."""
+        return self._reps
+
+    @property
+    def evolution(self) -> EvolutionBase:
+        """The evolution converter used to compute the evolution."""
+        return self._evolution
+
+    @property
+    def operators(self) -> List[OperatorBase]:
+        """The operators that are evolved in this circuit."""
+        return self._operators
+
+    @operators.setter
+    def operators(self, operators: Union[OperatorBase, List[OperatorBase]]) -> None:
```

Comment:
Shouldn't setting the operators invalidate the built data?

At this point one can manage the set of operators being used a whole set. I guess we can see what the sort of code looks like for one where the set of operators can be managed over an active/inactive list. Maybe as has been stated this is enough functionality to warrant another class deriving from this. I just wonder would we derive all our present evolved operator ansatzes from that and never directly from this. Can see how it pans out I guess.